File: Fortran/3_status.f90
program status
    implicit none
    include "mpif.h"
    
    INTEGER :: MyRank, ierror, num
    INTEGER :: stats(MPI_Status_Size)

    CALL MPI_Init(ierror)

    CALL MPI_Comm_rank(MPI_COMM_WORLD, MyRank, ierror)

    IF (MyRank == 0) THEN
        num = 309845
        CALL MPI_Send(num, 1, MPI_INT, 1, 101, MPI_COMM_WORLD, ierror)
        ! Size = 1, cause it's the size of the number of elements in an array, not the storage size of the variable
        PRINT *, "Sender side Rank = ", MyRank
    
    ELSE IF (MyRank == 1) THEN
        CALL MPI_Recv(num, 1, MPI_INT, MPI_ANY_SOURCE, MPI_ANY_TAG, MPI_COMM_WORLD, stats, ierror)
        PRINT *, "Receiver side Rank =", MyRank, ", Num = ", num , ", Source = ", stats(MPI_SOURCE), ", Tag = ",stats(MPI_TAG)
        PRINT *, "Error Status = ", stats(MPI_ERROR)
    END IF

    CALL MPI_Finalize(ierror)
end program status

File: C/3_status.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
    int rank, num;
    MPI_Status status;

    // MPI_Initialization
    MPI_Init(&argc, &argv);

    MPI_Comm_rank(MPI_COMM_WORLD, &rank);

    // code here
    if(rank==3)
    {
        num =30;
        MPI_Send(&num, 1, MPI_INT, 1, 101, MPI_COMM_WORLD);  // Thus the element size doesn't mean character length, rather number of elements out of an array
        printf("Sender side Rank %d: Num=%d \n", rank, num);
    }

    if(rank==1)
    {
        // use of wildcards at receiver side
        MPI_Recv(&num, 1, MPI_INT, MPI_ANY_SOURCE, MPI_ANY_TAG, MPI_COMM_WORLD, &status);
        printf("Receiver side Rank %d : Num=%d, Source=%d, Tag=%d \n Errorstatus=%d \n", rank, num, status.MPI_SOURCE, status.MPI_TAG, status.MPI_ERROR);
    }

    MPI_Finalize();
    return 0;
}